src/cript/nodes/exceptions.py

```diff
@@ -39,3 +39,25 @@ def __init__(self, node_type: str, json_dict: str):
 
     def __str__(self):
         return f"JSON Serialization failed for node type {self.node_type} with JSON dict: {self.json_str}"
+
+
+class UneditableAttributeError(Exception):
+    """
+    This exception is raised when the user attempts to edit a read only field
```

Comment:
I personally think this error would be more user friendly, especially since our data model and platform is a bit complex, but I can take it out for now and we can come and update it in the future if we need it

Reply:
we can have both if we overload the `__setattr__`

Reply:
I am a big advocate for accurate, understandable, and timely user feedback.

I updated the code and removed all the setters to for `Group` and `User` nodes for now. This will trigger an `AttributeError` if the user attempts to set it. For now, I think this is good enough, so that we can play around with it and debug it, but in future updates we probably want to come back and update all read-only attributes to be more user friendly.

I wrote an [issue](https://github.com/C-Accel-CRIPT/Python-SDK/issues/45) for it so we do not forget about it.